cognee/tasks/storage/index_data_points.py

```diff
@@ -30,7 +33,10 @@ async def index_data_points(data_points: list[DataPoint]):
 
     for index_name, indexable_points in index_points.items():
         index_name, field_name = index_name.split(".")
-        await vector_engine.index_data_points(index_name, field_name, indexable_points)
+        try:
+            await vector_engine.index_data_points(index_name, field_name, indexable_points)
+        except (OpenAIError, BadRequestError) as e:
```

Comment:
Do we except these errors because the embeddings call can fail?
If so, I would prefer making a custom exception like `EmbeddingException` for example that we can catch here and handle. `OpenAIError` and `BadRequestError` are implementation details of the vector engine and should not be exposed.

Reply:
Sounds reasonable. I made some changes. Can you please check if this addresses it?